api/src/main/java/org/apache/cloudstack/api/BaseListCmd.java

```diff
@@ -94,7 +94,7 @@ public Long getPageSizeVal() {
         if (pageSizeInt != null) {
             defaultPageSize = pageSizeInt.longValue();
         }
-        if (defaultPageSize.longValue() == s_pageSizeUnlimited) {
+        if (defaultPageSize!= null && defaultPageSize.longValue() == s_pageSizeUnlimited) {
```

Comment:
```suggestion
        if (s_pageSizeUnlimited.equals(defaultPageSize) {
```
just being a pain here

Reply:
This is fine Daan, this code came in as part of some unrelated merge.